### DISCUSSION TITLE
Importing flake.homeModule in home-manager results in unconfigured neovim

### DISCUSSION THREAD

**Discussion by MATOO-Dev** [BirdeeHub/nixCats-nvim#68] [Q&A] [answered]
I've setup a basic config with some plugins, and I'm now trying to add it as a home-manager module to replace my current nixvim setup.

I added my config flake as `nvim-config` in my flake inputs, and added `inputs.nvim-config.homeModule` to my home-manager imports

This builds successfully, but running `nvim` results in a default, unconfigured neovim

I've also tried `inputs.nvim-config.packages.x86_64-linux.nvim`, in which case the build fails with `The option '__ignoreNulls' does not exist`

What is the correct way to include my config in home-manager?
Thanks!

**Comment by BirdeeHub** [comment]
did you enable it? Putting homeModule in your home manager imports would import the module but you also have to enable it. And if you already have an nvim installed home manager will yell at you. Here are the module options available

https://nixcats.org/nixCats_hm_options.html

The module inherits the packages from your flake and also allows you to modify them further.

Did you override luaPath in the module? If so it would be looking in the new location for the config which would be wrong.

### as far as the other thing you tried:

This is a package not a module.

```nix
# no
imports = [
inputs.nvim-config.packages.x86_64-linux.nvim
];
# yes
home.packages = [
inputs.nvim-config.packages.x86_64-linux.nvim
];
```

So yeah, __ignoreNulls is not an option, its an attribute of a derivation that you are feeding into your modules list

**Comment by BirdeeHub** [comment]
Also if you could put further questions like this in discussions, I would appreciate it as then maybe other people will find it

Edit: thanks, I can move it apparently, who knew!

**Comment by MATOO-Dev** [comment]
> did you enable it? Putting homeModule in your home manager imports would import the module but you also have to enable it.

The flake is copied from the template, so `defaultPackageName = "nvim"`

I added `nvim.enable = true` to my home-manager config, but that did not seem to have any effect

> Did you override luaPath in the module? If so it would be looking in the new location for the config which would be wrong.

I didn't

> Also if you could put further questions like this in discussions, I would appreciate it as then maybe other people will find it

Noted, thanks for the correction. There should be a button on the right to allow you to convert the issue to a discussion if you want.

EDIT: 
I tried the other way of adding it to packages directly and that worked, ideally I'd still like to understand how to use it as a home-manager module though

**Reply by BirdeeHub** [reply, marked as answer]
You should be able to:

```nix
{ config, pkgs, lib, inputs }: {
  imports = [ inputs.yournvimflake.homeModule ];
  config = {
    nvim.enable = true;
    # nvim.packageNames = [ "nvim" ]; # <-- this should already be the default value if defaultPackageName is nvim
    # nvim.nixpkgs_version = inputs.nixpkgs; # <- optional
  };
}
```

assuming your defaultPackageName passed to mkHomeModules at the end of your template is "nvim"

I use it in my personal config in this manner and the tests also test module import so it should work if you are importing it correctly

**Reply by BirdeeHub** [reply]
There isnt a lot of info here for me to go on to provide more help

> Noted, thanks for the correction. There should be a button on the right to allow you to convert the issue to a discussion if you want.

I didnt know about this thank you!

**Reply by MATOO-Dev** [reply]
Explicitly adding `nvim.packagesNames = [ "nvim" ];` did the trick, despite already being the default.
Thanks for your help!

**Reply by BirdeeHub** [reply]
huh....

Weird.

Well, its a good idea to do anyway so that you can install any extra packages if you want like a notes specific one or to have your one with an unwrapped config for quick edits also installed to your profile.

But yeah I might have to look into that or just mark it as always required...

**Reply by BirdeeHub** [reply]
the tests seem to agree with you... When I stop setting that in the home manager test module it seems to not work...

**Reply by BirdeeHub** [reply]
maybe I spoke too soon and it should have been in issues XD but regardless now I know that I can just wait and if it ends up being for discussions I can move it over without needing to tell people to put it in discussions so thank you for that.

**Reply by BirdeeHub** [reply]
> Explicitly adding `nvim.packagesNames = [ "nvim" ];` did the trick, despite already being the default. Thanks for your help!

I think I just found out why....

Edit: I did :)

https://github.com/BirdeeHub/nixCats-nvim/commit/b1dd5412fc4fa2c06aaa36436e75cda8a34219d0

**Reply by BirdeeHub** [reply]
also my apologies about the tone of my original reply, upon re-reading it, it came off ruder than intended.

**Reply by MATOO-Dev** [reply]
> also my apologies about the tone of my original reply, upon re-reading it, it came off ruder than intended.

No worries, thanks for your help! ^^
